mlflow/models/evaluation/base.py

```diff
@@ -456,11 +485,99 @@ def _start_run_or_reuse_active_run(run_id):
     else:
         if run_id and active_run.info.run_id != run_id:
             raise ValueError(
-                "An active run exists, you cannot specify another run_id when " "evaluating."
+                "An active run exists, you cannot specify another run_id when evaluating."
             )
         yield active_run.info.run_id
 
 
+def _normalize_evaluators_and_evaluator_config_args(
+        evaluators,
+        evaluator_config,
+):
+    from mlflow.models.evaluation.evaluator_registry import _model_evaluation_registry
+
+    if evaluators is None:
+        evaluator_name_list = list(_model_evaluation_registry._registry.keys())
+        if evaluator_config is not None:
+            raise ValueError(
+                'If `evaluators` argument is None, `evaluator_config` argument must be None too.'
+            )
+        evaluator_name_to_conf_map = {}
+    elif isinstance(evaluators, str):
+        if not (evaluator_config is None or isinstance(evaluator_config, dict)):
+            raise ValueError(
+                "If `evaluators` argument is the name of an evaluator, evaluator_config must be "
+                "None or a dict containing config items for the evaluator."
+            )
+        evaluator_name_list = [evaluators]
+        evaluator_name_to_conf_map = {evaluators[0]: evaluator_config}
+    elif isinstance(evaluators, list):
+        if evaluator_config is not None:
+            if not (
+                    isinstance(evaluator_config, dict)
+                    and all(k in evaluators and isinstance(v, dict)
+                            for k, v in evaluator_config.items())
+            ):
+                raise ValueError(
+                    "If `evaluators` argument is a evaluator name list, evaluator_config "
+                    "must be a dict contains mapping from evaluator name to individual "
+                    "evaluator config dict."
+                )
+        evaluator_name_list = list(set(evaluators))
+        evaluator_name_to_conf_map = evaluator_config or {}
+    else:
+        raise ValueError(
+            '`evaluators` argument must be None, a evaluator name string, or a list of '
+            'evalautor names.'
+        )
+
+    return evaluator_name_list, evaluator_name_to_conf_map
+
+
+def _evaluate(
+        model, model_type, dataset, actual_run_id, evaluator_name_list, evaluator_name_to_conf_map
+):
+    """
+    This method is the patch point for databricks instrumentation.
+    The public API "evaluate" will verify argument first, and then pass normalized arguments
+    to the _evaluate method.
+    """
+    # import _model_evaluation_registry and PyFuncModel inside function to avoid circuit importing
+    from mlflow.models.evaluation.evaluator_registry import _model_evaluation_registry
+
+    client = mlflow.tracking.MlflowClient()
+    model_uuid = model.metadata.model_uuid
```

Comment:
For older pyfunc models that don't have a UUID attribute, this will change every time the model is reloaded for evaluation, right? Could we make the UUID stable for older models by taking a hash of the MLmodel file as the uuid when `Model.load()` is called?

Reply:
I create a PR to make older version loaded model the "model_uuid" attribute to be None.
Shall we directly log the model_uuid as `None` for older mlflow model ?

Reply:
Sounds good to me! I left a comment on the PR, but the approach sounds good: https://github.com/mlflow/mlflow/pull/5167/files#r769374910.